server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/routes/JMAPApiRoutes.scala

```diff
@@ -35,24 +35,29 @@ import org.apache.james.jmap.exceptions.UnauthorizedException
 import org.apache.james.jmap.http.Authenticator
 import org.apache.james.jmap.http.rfc8621.InjectionKeys
 import org.apache.james.jmap.json.Serializer
-import org.apache.james.jmap.method.CoreEcho
+import org.apache.james.jmap.method.Method
 import org.apache.james.jmap.model.Invocation.{Arguments, MethodName}
 import org.apache.james.jmap.model.{Invocation, RequestObject, ResponseObject}
 import org.apache.james.jmap.{Endpoint, JMAPRoute, JMAPRoutes}
+import org.apache.james.mailbox.MailboxSession
 import org.slf4j.{Logger, LoggerFactory}
 import play.api.libs.json.{JsError, JsSuccess, Json}
 import reactor.core.publisher.Mono
 import reactor.core.scala.publisher.{SFlux, SMono}
 import reactor.core.scheduler.Schedulers
 import reactor.netty.http.server.{HttpServerRequest, HttpServerResponse}
 
+import scala.collection.mutable
+import scala.jdk.CollectionConverters._
+
 object JMAPApiRoutes {
   val LOGGER: Logger = LoggerFactory.getLogger(classOf[JMAPApiRoutes])
 }
 
 class JMAPApiRoutes @Inject() (@Named(InjectionKeys.RFC_8621) val authenticator: Authenticator,
-                               val serializer: Serializer) extends JMAPRoutes {
-  private val coreEcho = new CoreEcho
+                               serializer: Serializer,
+                               javaMethods: java.util.Set[Method]) extends JMAPRoutes {
+  private val methods: mutable.Set[Method] = javaMethods.asScala
```

Comment:
you probably don't want it to be mutable, right?

Reply:
probably not, but that's why the conversion with `asScala` is doing, didn't find better at that moment. Idea?